anyway/widgets/suburban_widgets/injured_count_by_accident_year_widget.py

```diff
@@ -28,6 +29,23 @@ def __init__(self, request_params: RequestParams):
             "Fatal, severe and light injured count in the specified years, split by injury severity"
         )
 
+    @staticmethod
+    def sort_by_years_and_severity(
+        data: defaultdict, years_range: typing.Iterable
+    ) -> Dict[int, dict]:
+        res = {}
+        for year in years_range:
+            res[year] = {
+                InjurySeverity.KILLED.value: 0,
+                InjurySeverity.SEVERE_INJURED.value: 0,
+                InjurySeverity.LIGHT_INJURED.value: 0,
+            }
+            severity_in_year = data.get(year, {})
+            for severity, count in severity_in_year.items():
+                res[year][severity] += count
+
+        return res
+

```

Comment:
It is not clear what is actually sorted in the structure that is returned by the method. I would prefer:

- Applying the sort only on the structure (usually will be a list) that is actually sorted.
- Use a general function to sort, that will receive the entities order (instead of a function for each widget, or for each entity). That function will use the `sort` function.

Reply:
I need not only to sort but also fill the gaps in each level (if in the dict has keys 2017, 2020 then i need to add 2018, 2019),
will change it to be more generic

Reply:
Doing both in same method will result in many similar methods in different widgets. Using general purpose functions for the fill and sort, will allow us to have this code once. A function that fills empty values already exists  - `widgets.widget_utils.add_empty_keys_to_gen_two_level_dict`.  `widgets.widget_utils.format_2_level_items` might also be handy, though it may needs to be adapted for the new format.